kornia/feature/matching.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import math
+import torch
+import torch.nn.functional as F
+import typing
+from typing import Tuple, Optional
+TupleTensor = Tuple[torch.Tensor, torch.Tensor]
+
+
+def match_nn(desc1: torch.Tensor, desc2: torch.Tensor) -> TupleTensor:
```

Comment:
in the lasts PRs I discouraged to use predefined custom types to improve code readability